bokeh/charts/chart.py

```diff
@@ -39,39 +40,38 @@
 from ..resources import INLINE
 from ..util.notebook import publish_display_data
 from ..util.serialization import make_id
+from ..util.future import with_metaclass
+from ..model import Viewable
+from ..themes import Theme
 
 #-----------------------------------------------------------------------------
 # Classes and functions
 #-----------------------------------------------------------------------------
 
-def _make_method(prop_name):
-    def method(self, value):
-        setattr(self._options, prop_name, value)
-        return self
-    method.__doc__ = """ Chained method for %s option.
-    """ % prop_name
-    return method
+Scale = enumeration('linear', 'categorical', 'datetime')
 
-def _chained_options(opts_type):
-    def wrapper(cls):
-        orig_init = cls.__init__
+class ChartsTheme(Theme):
+    def apply_to_model(self, model):
+        """Apply this theme to a model. Don't call this method directly,
+        instead set the theme on the Document the model is a part of."""
 
-        cls_props = set(cls.properties())
+        if not isinstance(model, Chart):
+            raise ValueError("ChartsTheme should be only used on Chart objects \
+            but it's being used on %s instead." % model)
 
-        def __init__(self, *args, **kwargs):
-            self._options = opts_type(**kwargs)
-            orig_init(self)
+        attrs = {}
+        for k in model.properties_with_values(include_defaults=True):
+            if hasattr(self, k):
+                attrs[k] = getattr(self, k)
 
-        cls.__init__ = __init__
+        if attrs:
+            self._by_class_cache = {}
+            self._json = {'attrs' : { 'Chart' : attrs }}
 
-        for prop_name in opts_type.properties():
-            if prop_name not in cls_props:
-                setattr(cls, prop_name, _make_method(prop_name))
+        model.apply_theme(self._for_class(model.__class__))
```

Comment:
> If you are doing this I think you could just use a dict - you don't need a subtype of Theme or a special class?
defaults = {} ought to work as far as I can see above, then model.apply_theme(defaults) below.

Correct. I could just use `dict`. I didn't because I want to hook with the theme chain even if it's basically empty in this case...

As a general thought I agree with you regards "having a non-charts specific" solution.. but this would require some time/efforts.

> (what I'm saying is that even as a short-term fix, we know of two bugs created here - Chart would ignore the global default theme, and charts.defaults would stop working as a side effect of adding the Chart to a Document)

Mmmm... I'd expect `Chart` wouldn't ignore `default_theme` since  `super(Chart, self).__init__(*args, **kwargs)` still calls `default_theme.apply_to_model(self)` on the `Model` constructure. I guess what I was missing is what you mentioned here:

> when the Chart is added to a Document, it will switch to Document.theme overwriting the above

... but in that case I don't understand why why the `line_multi.py` charts example seems to work with this defaults thing.